classes/class.ilAdobeConnectXMLAPI.php

```diff
@@ -2095,4 +2099,50 @@ public function getTemplates($pluginObj)
 		asort($templates);
 		return $templates;	
 	}
-}
\ No newline at end of file
+
+
+	protected function proxy($ctx = null)
+	{
+	
+		require_once('Services/Http/classes/class.ilProxySettings.php');
+
+		if( ilProxySettings::_getInstance()->isActive() )
```

Comment:
What if the proxy is disabled? Returning _void_ would result in $ctx = null; in the consuming code.

Reply:
Your are completely right.

return null;
at the end of the function should fix it. [file_get_contents()](http://php.net/manual/de/function.file-get-contents.php) can deal with $ctx = null and $this->proxy() will not result in any changes. I could not test this case (DFN blocks foreign ip addresses).

Reply:
But shouldn't we return a valid stream context if the proxy is not used?

Let's assume the following code:
```
$context = array(
	'http'  => array(	
		'timeout' => 4	
	),
	'https' => array(
		'timeout' => 4
	)
);

$ctx = $this->proxy($context);
$xml_string = file_get_contents($url, false, $ctx);
```


If the proxy is not used, the timeout settings defined in the consuming code will not be applied and NULL will be returned. This could result in a timeout behaviour change.

Reply:
Good point!

Reply:
How about the last commit?
Lookup for defined context:

```php
if( ilProxySettings::_getInstance()->isActive() ) {
 # ....
} elseif( is_array($ctx) && count($ctx) ) {
      return stream_context_create($ctx);
}

return null;
```

Reply:
Yes, this looks good.